src/Microsoft.OData.Core/Json/ODataJsonWriterUtilsAsync.cs

```diff
@@ -0,0 +1,274 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataJsonWriterUtilsAsync.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Json
+{
+    #region Namespaces
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.OData.JsonLight;
+    #endregion Namespaces
+
+    /// <summary>
+    /// Helper methods used by the OData writer for the JSON format.
+    /// </summary>
+    internal static partial class ODataJsonWriterUtils
+    {
+        /// <summary>
+        /// Asynchronously writes an error message.
+        /// </summary>
+        /// <param name="jsonWriter">The JSON writer to write the error.</param>
+        /// <param name="writeInstanceAnnotationsDelegate">Delegate to write the instance annotations.</param>
+        /// <param name="error">The error instance to write.</param>
+        /// <param name="includeDebugInformation">A flag indicating whether error details should be written (in debug mode only) or not.</param>
+        /// <param name="maxInnerErrorDepth">The maximum number of nested inner errors to allow.</param>
+        /// <param name="writingJsonLight">true if we're writing JSON lite, false if we're writing verbose JSON.</param>

```

Comment:
Why not have `verbose` as the param?

Reply:
Also if you change it to `verbose` the condition may need to be flipped around

Reply:
@wandeg While your suggestion may be valid, I'd lean towards letting the name be so as not to deviate much from the already existing synchronous equivalent defined [here](https://github.com/OData/odata.net/blob/25ea344714d291d169b48ebf71c09721014989aa/src/Microsoft.OData.Core/Json/ODataJsonWriterUtils.cs#L39).
In addition, `verbose` also doesn't really tell us much about what happens when the value is `true` or `false`.
A user would need to remember that its called one thing when invoking the synchronous method and another when invoking the asynchronous equivalent.
[Here's](https://github.com/OData/odata.net/blob/25ea344714d291d169b48ebf71c09721014989aa/src/Microsoft.OData.Core/Json/ODataJsonWriterUtils.cs#L202) how the flag is applied:
```csharp
if (writingJsonLight)
{
    Debug.Assert(writeInstanceAnnotationsDelegate != null, "writeInstanceAnnotations != null");
    writeInstanceAnnotationsDelegate(instanceAnnotations);
}
```
When the `writingJsonLight` is true, we invoke a delegate to write instance annotations

Reply:
For the async methods, can we just remove the verbose option?  That is a carry-over from OData V3 -- new code should never be writing verbose (opportunity for 8.0 clean-up to synchronous methods).

---
In reply to: [599501467](https://github.com/OData/odata.net/pull/2031#discussion_r599501467) [](ancestors = 599501467)

Reply:
@mikepizzo @gathogojr 
We can create an issue for tracking v8.0 cleanup items

Reply:
@mikepizzo To confirm I understood correctly, we use the `writingJsonLight` flag as a condition to invoke the delegate for writing instance annotations. If we get rid of that flag, we get rid of that delegate as well, right?

Reply:
We should *always* write as if writingJsonLight was true.